nemo/collections/multimodal/modules/common/audio_text_generation_utils.py

```diff
@@ -0,0 +1,765 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for generating text."""
+
+import pickle
+from collections.abc import Iterable
+from functools import partial
+from typing import Callable, Tuple
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+from nemo.collections.common.tokenizers.tabular_tokenizer import TabularTokenizer
+from nemo.collections.multimodal.modules.common.audio_text_generation_strategy import (
+    END_OF_SEQ,
+    model_inference_strategy_dispatcher,
+)
+from nemo.collections.nlp.modules.common.megatron.utils import get_ltor_masks_and_position_ids
+from nemo.collections.nlp.modules.common.transformer.text_generation import LengthParam, OutputType, SamplingParam
+from nemo.utils import AppState
+
+try:
+    from apex.transformer.pipeline_parallel.utils import _reconfigure_microbatch_calculator
+
+    HAVE_APEX = True
+
+except (ImportError, ModuleNotFoundError):
+
+    HAVE_APEX = False
+
+try:
+    from megatron.core import parallel_state, tensor_parallel
+
+    HAVE_MEGATRON_CORE = True
+
+except (ImportError, ModuleNotFoundError):
+
+    HAVE_MEGATRON_CORE = False
+
+__all__ = [
+    "get_default_sampling_params",
+    "get_default_length_params",
+    "get_computeprob_response",
+    "generate",
+]
+
+
```

Comment:
Just figured that we don't need all these functions here, some can be directly imported from `nemo.collections.nlp.modules.common.text_generation_utils`: 
```
get_default_sampling_params, 
get_default_length_params, 
get_computeprob_response, 
tab_logits,
top_k_logits,
repetition_penalty,
get_model_parallel_src_rank,
switch,
tab_sample_sequence_batch,
sample_token_greedy,
sample_token_topk,
sample_token_topk_beam_search,
compute_beam_search_len_penalty,
get_sampling_token_fn,

Reply:
y removed unnecessary part